### PR TITLE
angular 7 upgrade

### PR DESCRIPTION
**Pull request by whyboris**
👍 App upgraded to `angular 7`

_no behavior should change for the user_

### CODE REVIEW

**Comment by whyboris** [comment]
Windows and Mac builds well. Have not tested Linux (and will not during this iteration - no system available to test it right now).

**Comment by whyboris** [comment]
On your PR #2 it worked until you merged the new _ffmpeg_ branch. I checked on a _Mac_ and saw the same issue (images not showing up). Unsure at what point it happened.

Without applying your branch - just looking at this branch alone, images work for you too, right?

File: electron-builder.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
   "mac": {
     "category": "public.app-category.video",
     "target": "dmg",
-    "icon": "dist/favicon",
+    "icon": "src/assets/icons.icns",
```
Comment (cal2195) [inline electron-builder.json]: Just a note, if you apply the same icon fix to linux, it builds for me! 👍 

I will test out the built image when I get home! 😄

**Comment by cal2195** [comment]
> On your PR #2 it worked until you merged the new _ffmpeg_ branch - I only checked on a _Mac_ and saw the same issue (images not showing up). Unsure at which new code makes it so.
> 
> Without applying your branch - just looking at this branch alone, images work for you too, right?

Nope, same issue with just this branch for me, although the old image names are being used! Are you not seeing it?

Maybe I have something cached...?

**Comment by whyboris** [comment]
Old image names = `1.jpg`, `2.jpg`, etc -- right? That's expected behavior as this branch hasn't pulled in any of your changes - it's just my _master_ getting upgraded to _angular 7_.

**Comment by cal2195** [comment]
Yep - I just meant that it's correctly using the old images, but not appending `file:///` even with just this branch!

**Comment by whyboris** [comment]
You'll need to `npm install` ty try it out ... it will update _A LOT_ of dependencies, so it's better to `git clone` into a new folder - so it doesn't mess with your branches too much 😃

**Comment by whyboris** [comment]
Oh man -- I see ... doesn't work for me on mac -- works on PC. Will have to investigate -- will fix 👍

**Comment by whyboris** [comment]
⚠️ Does not prepend `file:///` on Mac - will fix on Friday when I'm home from work 👍

**Comment by whyboris** [comment]
The app does work on _Mac_ once you build it ... it just fails to show images in _dev_ mode -- probably some setting in _Electron_ that requires you enable local file displaying.

I had to put this in this when I started working on Windows: 
```ts
process.env['ELECTRON_DISABLE_SECURITY_WARNINGS'] = 'true';
```
Maybe Mac needs some other thing too 😖

File: main.ts [hunk 2]
```diff
@@ -115,7 +117,17 @@ function createWindow() {
 
   // Open the DevTools.
   if (serve) {
+    require('electron-reload')(__dirname, {
+      electron: require(`${__dirname}/node_modules/electron`)
+    });
+    win.loadURL('http://localhost:4200');
```
Comment (cal2195) [inline main.ts]: My guess is this! 😉

File: main.ts [hunk 3]
```diff
@@ -115,7 +117,17 @@ function createWindow() {
...
     win.webContents.openDevTools();
+  } else {
+    win.loadURL(url.format({
+      pathname: path.join(__dirname, 'dist/index.html'),
+      protocol: 'file:',
```
Comment (cal2195) [inline main.ts]: As this uses file! 😄

**Comment by whyboris** [comment]
Ready for re-review ✅

File: tsconfig.json [hunk 4]
```diff
@@ -2,31 +2,26 @@
...
-    ],
     "typeRoots": [
       "node_modules/@types"
     ],
     "lib": [
       "es2016",
       "dom"
-    ]
+    ],
+    "module": "es2015",
```
Comment (cal2195) [inline tsconfig.json]: This trailing comma is causing an error in Atom! Please remove! 😄

File: electron-builder.json (hunk 1 above)
Comment (cal2195) [reply electron-builder.json]: Don't forget to change this! 👍

**Comment by whyboris** [comment]
Sorry unsure what I'm supposed to change about this:
```
-    "icon": "dist/favicon",
+    "icon": "src/assets/icons.icns",
```

**Comment by whyboris** [comment]
On my Mac the _.dmg_ file is 88.2MB, but when installed it's around 245MB. Unsure what happened on your Mac 😕

**Comment by cal2195** [comment]
> Sorry unsure what I'm supposed to change about this:
> 
> ```
> -    "icon": "dist/favicon",
> +    "icon": "src/assets/icons.icns",
> ```

Sorry, I meant duplicate this change to the linux one! 😄

**Pull request merged**